rapid7/metasploit-framework
# Detection module for CVE-2023-21554 - QueueJumper

Hi,

this module detects Windows hosts that are vulnerable to CVE-2023-21554 aka QueueJumper, as described in [https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-21554](https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-21554). So far no exploit exists yet, however the primitive was very well written up in [https://securityintelligence.com/posts/msmq-queuejumper-rce-vulnerability-technical-analysis/](https://securityintelligence.com/posts/msmq-queuejumper-rce-vulnerability-technical-analysis/) including a way to detect the vulnerability.

First the module sends a normal message packet to check if MSMQ is present. If that is the case, the module sends a malformed packet and checks whether a response is sent back or not. A patched MSMQ instance will not respond to a malformed packet.

Resolves #18242 

## Testing Environment Setup

- Install a Windows version which don't have any of the security updates mentioned on [https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-21554](https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-21554) or newer installed.
- Install MSMQ by executing the following command in an elevated Powershell prompt: `Install-WindowsFeature -Name MSMQ -IncludeAllSubFeature`
- In "services.msc" check that "Message Queuing" is running and port 1801 is open.

## Verification

List the steps needed to make sure this thing works

- [x] `use auxiliary/scanner/msmq/cve_2023_21554_queuejumper`
- [x] `set RHOSTS [HOST]`
